packages/gatsby-cli/README.md

```diff
@@ -32,10 +32,33 @@ You can also use the `package.json` script variant of these commands, typically
 
 ### `new`
 
-`gatsby new gatsby-site`
+```bash
+gatsby new [<site-name> [<starter-url>]]
+```
+
+#### Arguments
+
+- **`site-name` :** Your Gatsby site name along with the path should be pass here`
+
+- **`starter-url` :** Mention the gatsby starter pack url or local path here.
+  See the [Gatsby starters](https://www.gatsbyjs.org/docs/gatsby-starters/) for more.
+  `Default` starter pack is [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default).
+
+> **NOTE :** Dont pass `.` or `./` or `` _(space)_ in the `site-name`. It will throw error of specifying name and it will stop the operation
+
+#### Examples
+
+- `gatsby new MY_AWESOME_GATSBY_SITE` : it will create a gatsby site with `MY_AWESOME_GATSBY_SITE` name in the current working directory using the default starter
+
+- `gatsby new MY_AWESOME_GATSBY_BLOG_SITE https://github.com/gatsbyjs/gatsby-starter-blog` : It will create a gatbsy site with name `MY_AWESOME_GATSBY_BLOG_SITE` and using [gatsby-starter-blog](https://www.gatsbyjs.org/starters/gatsbyjs/gatsby-starter-blog/) starter pack
+
+- `gatsby new https://github.com/gatsbyjs/gatsby-starter-blog` : It will result in error and will stop the command with a message : _It looks like you forgot to add a name for your new project. Try running instead "gatsby new new-gatsby-project https://github.com/gatsbyjs/gatsby-starter-blog"_
```

Comment:
i would not suggest error messages as examples. insted do here the interactive

```suggestion
- If you leave all the both of the argument, then CLI will then run a interactive shell asking for these inputs:
``bash
gatsby new
? What is your project called? › my-gatsby-project  
? What starter would you like to use? › - Use arrow-keys. Return to submit. 
❯  gatsby-starter-default
   gatsby-starter-hello-world
   gatsby-starter-blog
   (Use a different starter)
``                                                                                                                
```

Reply:
What about the error message statement

Reply:
examples are here to show "how to use it" and not to show "how you use it wrong". i would not put the error message. to use the cli you don't need to know when an error check is available.

Reply:
Yeah I removed them

Reply:
👍